pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -86,14 +149,43 @@ func createRawDeployment(componentMeta metav1.ObjectMeta,
 	setDefaultDeploymentSpec(&deployment.Spec)
 	return deployment
 }
+func createRawWorkerDeployment(componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec, predictorName string, replicas int32) *appsv1.Deployment {
+	podMetadata := componentMeta
+	workerPredictorName := constants.GetRawWorkerServiceLabel(predictorName)
+	podMetadata.Labels["app"] = workerPredictorName
+	setDefaultPodSpec(podSpec)
+	deployment := &appsv1.Deployment{
+		ObjectMeta: componentMeta,
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": workerPredictorName,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: podMetadata,
+				Spec:       *podSpec,
+			},
+		},
+	}
+	if componentExt.DeploymentStrategy != nil {
+		deployment.Spec.Strategy = *componentExt.DeploymentStrategy
```

Comment:
I guess here we need to be a little cautious since GPU resources are often limited, and use rolling update might cause errors while spinning up the new pods while the old still running.

Reply:
Correct but it is a typical upgrade strategy to reduce down-time. In cases where a large number of resources are utilized, it’s important to employ the appropriate strategy. This seems to be more related to the infrastructure. Therefore, I think it would be beneficial to include this in the Considerations section of the documentation.